File: app/src/main/java/com/example/punchpad2/DraftManager.java
package com.example.punchpad2;

import android.content.Context;
import android.content.SharedPreferences;

public class DraftManager {

    private final SharedPreferences prefs;

    public DraftManager(Context context) {
        prefs = context.getSharedPreferences("Drafts", Context.MODE_PRIVATE);
    }

    public void saveDraft(String key, String content) {
        prefs.edit().putString(key, content).apply();
    }

    public String loadDraft(String key) {
        return prefs.getString(key, null);
    }

    public void clearDraft(String key) {
        prefs.edit().remove(key).apply();
    }
}

File: app/src/main/java/com/stratonotes/DraftManager.kt
package com.stratonotes

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit

class DraftManager(context: Context) {
    private val prefs: SharedPreferences =
        context.getSharedPreferences("Drafts", Context.MODE_PRIVATE)

    fun saveDraft(key: String?, content: String?) {
        prefs.edit() { putString(key, content) }
    }

    fun loadDraft(key: String?): String? {
        return prefs.getString(key, null)
    }

    fun clearDraft(key: String?) {
        prefs.edit() { remove(key) }
    }
}